src/common/src/hash/dispatcher.rs

```diff
@@ -58,6 +58,34 @@ pub trait HashKeyDispatcher {
     }
 }
 
+pub fn hash_key_data_size(data_type: &DataType) -> DataSize {
+    use std::mem::size_of;
+
+    use crate::types::{
+        Decimal, IntervalUnit, NaiveDateTimeWrapper, NaiveDateWrapper, NaiveTimeWrapper,
+        OrderedF32, OrderedF64,
+    };
+
+    match data_type {
+        DataType::Boolean => DataSize::FixedBits(1),
```

Comment:
1. We should also move `DataSize` and rename it to `HashKeySize`?
2. After 1, boolean should be `Fixed(1)`

Reply:
about 2
how can we distinguish between 1 bytes and  1 bit ?

Reply:
fix above